azure-kusto-data/azure/kusto/data/_token_providers.py

```diff
@@ -60,14 +60,14 @@ class TokenProviderBase(abc.ABC):
     _cloud_info = None
     _scopes = None
 
-    def __init__(self, kusto_uri: str):
+    def __init__(self, is_async: bool, kusto_uri: str):

```

Comment:
Consider add a default value so you won't need to pass 'False' it when using sync

Reply:
The reasons I made it this way:
1. I prefer to keep it explicit, to minimize mistakes, especially since it's internals to it won't make customer's code more verbose
2. If it has a default value then it has to be at the end, right now it's the first argument in all of the token providers, so it's more consistent, and we won't have to change the code when adding parameters to child providers

Reply:
nit:
For this and all derived classes , not a fan on the position of this argument as the first.
It also conflicts with the order of this argument in AAD Helper

Reply:
Where would you put in `MsiTokenProvider`
`def __init__(self, is_async: bool, kusto_uri: str, msi_args: dict = None)`?

Reply:
interesting.
I would either make it optional or use typing to state msi_args can be dict or none.
This is an internal class, so not critical